### ISSUE TITLE
RDP Sessions disappear suddenly in Wayland when switching Workspace

### ISSUE THREAD

**Issue by chnusti** [FreeRDP/Remmina#1295]
We support Remmina 1.2+ versions only.
Please do not report bugs occurring on older versions!

Please give us some details about the systems you are using:
* Client (OS name and version): Fedora 26
* Remmina version (remmina --version): 1.2.0-rcgit-20
* Desktop environment (GNOME, Unity, KDE, ..): Gnome-Shell
* Connecting to (OS and version): Windows 7
* Connecting via (RDP, VNC, ...): RDP
* RDP issues only, freerdp version (xfreerdp --version): 

As written in the subject during work on a different workspace and wanna switch back to my rdp session the window disappear suddenly.
On the other hand the session is still in use when looking lsof:
rn    6035 19223    47u     IPv4            1789878       0t0        TCP 1.2.3.4:40596->f.f.o:3389 (ESTABLISHED)
rn    6035 19224    47u     IPv4            1789878       0t0        TCP 1.2.3.4:40596->f.f.o:3389 (ESTABLISHED)
rn    6035 19225    47u     IPv4            1789878       0t0        TCP 1.2.3.4:40596->f.f.o:3389 (ESTABLISHED)
rn    6035 19226    47u     IPv4            1789878       0t0        TCP 1.2.3.4:40596->f.f.o:3389 (ESTABLISHED)
rn    6035 19227    47u     IPv4            1789878       0t0        TCP 1.2.3.4:40596->f.f.o:3389 (ESTABLISHED)
rn    6035 19228    47u     IPv4            1789878       0t0        TCP 1.2.3.4:40596->f.f.o:3389 (ESTABLISHED)

And when I reconnect the message "Another user is connected to the server...." is shown....

**Comment by antenore** [comment]
I've already seen this issue, but on Red Hat EL 6.1 and Gnome as DE, but we have never found what is causing it. It looks like the window is outside the screen area and therefore not visible.

 He finally switched to xfce4, bit of course I'm not advising to do the same. 

As I cannot reproduce the problem I don't know how to help out.

I'd probably use ```xprop``` to get the window id of remmina before it disappears, and than I'd use xdotool or other command line tools to move it in the right screen/workspace.

For the next release (it'll be quite soon hopefully), we have solved some gnome-shell related bugs and integration issues that may help.

Try also to execute remmina from a terminal and see if there are any meaningful massages that may help us.

**Comment by chnusti** [comment]
Nope. I've already tried to start remmina from terminal but there is no output at all.
On the other hand if the window would be out of screen I should still see the window somewhere. like in the open programs bar, isn't it?

However, this is pretty annoying.

**Comment by antenore** [comment]
Sure it is, I've the same setup and I'm testing

```
$ lsb_release -d
Description:    Fedora release 26 (Twenty Six)
$ remmina --version
Remmina - Version 1.2.0-rcgit-20 (git 25e4b13b)
$ xfreerdp --version
This is FreeRDP version 2.0.0-dev (git f9a47a346)
```

The only difference is that I compile freerdp and remmina by myslef

Are you using Wayland? How did you install remmina and freerdp?

**Comment by chnusti** [comment]
yes, I'm using wayland.
I installed freerdp from fedora repo.
probably I could try to install from source?

**Comment by madscientist** [comment]
As I mentioned in a comment to issue #1353 I see this constantly; very simple to reproduce.  As with the user above, I don't think the window is just moved somewhere because I can't see it in my window lists at all which I should be able to do even if it's at some weird geometry.

I'm using Ubuntu 17.10 Gnome (not Unity) with Xorg (not Wayland) with the latest build from the PPA: Remmina 1.2.0-rcgit-27, with xfreerdp 1.1.0-beta1

**Comment by madscientist** [comment]
For what it's worth I added the freerdp PPA http://ppa.launchpad.net/remmina-ppa-team/freerdp-daily/ubuntu and upgraded to libfreerdp2 2.0.0~git201803011108, and now this problem seems have disappeared for me (using Remmina 1.2.0-rcgit-27-0~remmina201802142315).

I left my Remmina window opened to a Windows system via RDP all night last night, on another workspace, and it's still just fine this morning.  So, maybe a freerdp issue with the older release?

**Comment by antenore** [comment]
It could be. The FreeRDP between November and end of January have done tons of modifications (for the good), and unlucky they have introduced several bugs. Now it seems it's quite stable and it should work properly.

I'll update the FreeRDP packages ASAP, today we discovered an important bug in Remmina, so soon we will have to release a new intermediate version.

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @antenore @chnusti @madscientist
